support/sier-codec/src/schema.rs

```diff
@@ -0,0 +1,245 @@
+use blake2::{Blake2b, Digest};
+use core::convert::TryInto;
+
+use crate::{Error, Object, Value};
+
+pub type Id = [u8; 8];
+
+pub struct StructDef {
+    pub(crate) type_name: String,
+    pub(crate) fields: Vec<FieldDef>,
+}
+
+impl StructDef {
+    pub fn type_name(&self) -> &str {
+        &self.type_name
+    }
+
+    pub fn id(&self) -> [u8; 8] {
+        let mut hasher = Blake2b::new();
+
+        hasher.update(&self.type_name);
+        for field in &self.fields {
+            hasher.update(&field.name);
+            hasher.update(field.type_.id());
+        }
+
+        let digest = hasher.finalize();
+
+        let bytes: &[u8] = &digest[0..8];
+        bytes.try_into().expect("hash should always be > 8 bytes")
```

Comment:
this is a panicing function, I remember from our conversations we want to avoid this in the blockchain code, is this lib ever going to be used in the blockchain implementation?

Reply:
Yes. They should be removed. I missed this one.

Reply:
This will probably not be used in the blockchain. I'm not aware of a way to get a fixed size array from a larger fixed size array without doing some type of unwrap.

I just found the Split impl (https://docs.rs/generic-array/0.14.4/generic_array/struct.GenericArray.html#impl-Split%3CT%2C%20K%3E) for generic array. I'll see if I can use that to not look like it may panic.

Reply:
It seems hard/impossible to convert a GenericArray to a fixed size array, even if the types line up. I have an approach that could work to make it not look like it might panic, but I don't think it's quite worth it ATM. Added TODO.

Reply:
one way of solving this could be to check on the byte size and return a Result based on this, I am not deep enough in the context of this function to determine the probability it will panic. If this panic cannot happen (from the structure or usage) this is not an issue, otherwise maybe do this size check.